src/instana/instrumentation/pep0249.py

```diff
@@ -2,139 +2,161 @@
+
+if TYPE_CHECKING:
+    from instana.span.span import InstanaSpan
 
 
 class CursorWrapper(wrapt.ObjectProxy):
-    __slots__ = ('_module_name', '_connect_params', '_cursor_params')
+    __slots__ = ("_module_name", "_connect_params", "_cursor_params")
 
-    def __init__(self, cursor, module_name,
-                 connect_params=None, cursor_params=None):
+    def __init__(
+        self, cursor, module_name, connect_params=None, cursor_params=None
+    ) -> None:
         super(CursorWrapper, self).__init__(wrapped=cursor)
         self._module_name = module_name
         self._connect_params = connect_params
         self._cursor_params = cursor_params
 
-    def _collect_kvs(self, span, sql):
+    def _collect_kvs(self, span, sql) -> InstanaSpan:
         try:
-            span.set_tag(ext.SPAN_KIND, 'exit')
-
-            db_parameter_name = next((p for p in ('db', 'database', 'dbname') if p in self._connect_params[1]), None)
+            span.set_attribute(SpanKind, "exit")
+
+            db_parameter_name = next(
+                (
+                    p
+                    for p in ("db", "database", "dbname")
+                    if p in self._connect_params[1]
+                ),
+                None,
+            )
             if db_parameter_name:
-                span.set_tag(ext.DATABASE_INSTANCE, self._connect_params[1][db_parameter_name])
-
-            span.set_tag(ext.DATABASE_STATEMENT, sql_sanitizer(sql))
-            span.set_tag(ext.DATABASE_USER, self._connect_params[1]['user'])
-            span.set_tag('host', self._connect_params[1]['host'])
-            span.set_tag('port', self._connect_params[1]['port'])
+                span.set_attribute(
+                    SpanAttributes.DB_NAME,
+                    self._connect_params[1][db_parameter_name],
+                )
+
+            span.set_attribute(SpanAttributes.DB_STATEMENT, sql_sanitizer(sql))
+            span.set_attribute(SpanAttributes.DB_USER, self._connect_params[1]["user"])
+            span.set_attribute("host", self._connect_params[1]["host"])
+            span.set_attribute("port", self._connect_params[1]["port"])
         except Exception as e:
             logger.debug(e)
         return span
 
-    def __enter__(self):
+    def __enter__(self) -> CursorWrapper:
         return self
 
-    def execute(self, sql, params=None):
-        tracer, parent_span, operation_name = get_tracer_tuple()
+    def execute(self, sql, params=None) -> None:
+        tracer, _, operation_name = get_tracer_tuple()
 
         # If not tracing or we're being called from sqlalchemy, just pass through
-        if (tracing_is_off() or (operation_name == "sqlalchemy")):
+        if tracing_is_off() or (operation_name == "sqlalchemy"):
             return self.__wrapped__.execute(sql, params)
 
-        with tracer.start_active_span(self._module_name, child_of=parent_span) as scope:
+        with tracer.start_as_current_span(self._module_name) as span:
```

Comment:
Is there a reason why parent span's span_context is not passed?

Reply:
Didn't see an implementation with context of parent_span but find one from urllip integration now. I'm fixin it.